creator-node/src/redis.js

```diff
@@ -1,36 +1,123 @@
+/**
+ * Exports a Singleton Redis client instance, with custom wallet write locking logic
+ */
+
+const Redis = require('ioredis')
+
 const config = require('./config.js')
 const { logger: genericLogger } = require('./logging')
-const Redis = require('ioredis')
+const { asyncRetry } = require('./utils')
 
 const redisClient = new Redis(config.get('redisPort'), config.get('redisHost'))
 
-const EXPIRATION = 60 * 60 * 2 // 2 hours in seconds
-class RedisLock {
-  static async setLock(key, expiration = EXPIRATION) {
-    genericLogger.info(`SETTING LOCK ${key}`)
-    // set allows you to set an optional expire param
-    return redisClient.set(key, true, 'EX', expiration)
-  }
+const WalletWriteLock = {
+  WALLET_WRITE_LOCK_EXPIRATION_SEC: 1800, // 30 min in sec
 
-  static async getLock(key) {
-    genericLogger.info(`GETTING LOCK ${key}`)
-    return redisClient.get(key)
-  }
+  VALID_ACQUIRERS: {
+    UserWrite: 'userWrite',
+    SecondarySyncFromPrimary: 'secondarySyncFromPrimary',
+    PrimarySyncFromSecondary: 'primarySyncFromSecondary'
+  },
 
-  static async acquireLock(key, expiration = EXPIRATION) {
-    genericLogger.info(`SETTING LOCK IF NOT EXISTS ${key}`)
-    const response = await redisClient.set(key, true, 'NX', 'EX', expiration)
-    return !!response
-  }
+  getKey: function (wallet) {
```

Comment:
this should probably be a private helper fn

Reply:
ya its an object but ya fixed